lib/fog/azurerm/storage.rb

```diff
@@ -3,10 +3,17 @@ module Fog
   module Storage
     # This class registers models, requests and collections
     class AzureRM < Fog::Service
-      requires :tenant_id
-      requires :client_id
-      requires :client_secret
-      requires :subscription_id
+      # Recognizes when creating management client
+      recognizes :tenant_id
+      recognizes :client_id
+      recognizes :client_secret
+      recognizes :subscription_id
+
+      # Recognizes when creating data client
+      recognizes :azure_storage_account_name
+      recognizes :azure_storage_access_key
+      recognizes :azure_storage_connection_string
+      recognizes :debug
```

Comment:
Attributes required for blob service, should not be class attributes.In this case, user will have to provide extra attributes every time when these attributes are not required and just want to create a storage account.